### DISCUSSION TITLE
How to trigger a slack alert when should_execute function returns False on a Schedule

### DISCUSSION THREAD

**Discussion by toddy86** [dagster-io/dagster#20009] [Q&A] [answered]
Hi, 

We have a schedule which runs on a cron schedule every morning at 8am. In addition, we have a function which we pass to `should_run` to decide if the schedule should actually run or not. 

What I'd like to do is to have a Slack alert sent each time the schedule ticks when the `should_run_logic` function returns `False`. 

I've taken a look into the Dagster Cloud Slack alerts, but it doesn't seem to provide an option for exactly this. 

Is this possible in Dagster? Or will we need to hand-roll our own Slack alert to run inside the `should_run_logic` function?

```
def should_run_logic(context: ScheduleEvaluationContext) -> bool:
    run_job: boolean = my_code_logic()
    return run_job

my_schedule = ScheduleDefinition(
    job=my_job,
    cron_schedule="0 8 * * *",
    should_execute=should_run_logic,
)
```

**Comment by garethbrickman** [comment, marked as answer]
Hey Todd, I also don't think there's any convenient native ways to do this right now so hand-rolling would be necessary.

**Reply by toddy86** [reply]
Thanks, Gareth. I'll start rolling
